apps/microtvm/riscv/template_project/microtvm_api_server.py

```diff
@@ -0,0 +1,285 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import fcntl
+import os
+import os.path
+import pathlib
+import select
+import shutil
+import subprocess
+import tarfile
+import time
+import re
+
+from tvm.micro.project_api import server
+
+
+PROJECT_DIR = pathlib.Path(os.path.dirname(__file__) or os.path.getcwd())
+
+
+MODEL_LIBRARY_FORMAT_RELPATH = "model.tar"
+
+
+IS_TEMPLATE = not os.path.exists(os.path.join(PROJECT_DIR, MODEL_LIBRARY_FORMAT_RELPATH))
+
+# Used this size to pass most CRT tests in TVM.
+WORKSPACE_SIZE_BYTES = 2 * 1024 * 1024
+
+CMAKEFILE_FILENAME = "CMakeLists.txt"
+
+# The build target given to make
+BUILD_TARGET = "build/main"
+
+
+class Handler(server.ProjectAPIHandler):
+    BUILD_TARGET = "build/main"
+
+    def __init__(self):
+        super(Handler, self).__init__()
+        self._proc = None
+
+    def server_info_query(self, tvm_version):
+        return server.ServerInfo(
+            platform_name="host",
+            is_template=IS_TEMPLATE,
+            model_library_format_path=""
+            if IS_TEMPLATE
+            else PROJECT_DIR / MODEL_LIBRARY_FORMAT_RELPATH,
+            project_options=[
+                server.ProjectOption(
+                    "verbose",
+                    optional=["build"],
+                    type="bool",
+                    default=False,
+                    help="Run make with verbose output",
+                ),
+                server.ProjectOption(
+                    "workspace_size_bytes",
+                    optional=["generate_project"],
+                    type="int",
+                    default=WORKSPACE_SIZE_BYTES,
+                    help="Sets the value of TVM_WORKSPACE_SIZE_BYTES.",
+                ),
+                server.ProjectOption(
+                    "toolchain_path",
+                    optional=["generate_project"],
+                    type="str",
+                    default="/opt/riscv",
+                    help="Sets the value of toolchain path.",
+                ),
+                server.ProjectOption(
+                    "target",
+                    optional=["generate_project"],
+                    type="str",
+                    default="riscv64-unknown-linux-gnu",
+                    help="Sets the value of target.",
+                ),
+                server.ProjectOption(
+                    "march",
+                    optional=["generate_project"],
+                    type="str",
+                    default="rv64gc",
+                    help="Sets the value of target architecture.",
+                ),
+            ],
```

Comment:
To compare how this performs on different vector length, we should make it configurable either by exposing `vlen` and `elen` as project options or by adding a generic `spike_extra_args` option to add any user defined arguments to the cmdline. See here: https://github.com/PhilippvK/microtvm-spike-template/blob/main/template_project/microtvm_api_server.py#L136-L149